src/hooks/worlds/useWorldBySlug.ts

```diff
@@ -2,21 +2,24 @@ import { useMemo } from "react";
 import Bugsnag from "@bugsnag/js";
 import { where } from "firebase/firestore";
 
-import { ALWAYS_EMPTY_ARRAY, FIELD_HIDDEN, FIELD_SLUG, PATH } from "settings";
+import {
+  ALWAYS_EMPTY_ARRAY,
+  DEFERRED,
+  FIELD_HIDDEN,
+  FIELD_SLUG,
+  PATH,
+} from "settings";
 
-import { WorldId, WorldSlug, WorldSlugLocation, WorldWithId } from "types/id";
-
-import { convertToFirestoreKey } from "utils/id";
+import {
+  MaybeWorldSlugLocation,
+  WorldId,
+  WorldSlug,
+  WorldWithId,
+} from "types/id";
 
 import { useLiveCollection } from "hooks/fire/useLiveCollection";
 
-export const useWorldBySlug = (
-  slugOrOptions?: WorldSlug | string | WorldSlugLocation
-) => {
-  const slug: string = convertToFirestoreKey(
-    typeof slugOrOptions === "string" ? slugOrOptions : slugOrOptions?.worldSlug
-  );
-
+export const useWorldBySlug = ({ worldSlug: slug }: MaybeWorldSlugLocation) => {
```

Comment:
> For the most part, I think getting rid of the HOCs is a mistake. 

The removal of these HOCs isn't a mistake. Adding a proper one is the way to go. These were a stop-gap and added needless complexity.

> Switching out a single argument function for something that takes options. 
Still a single argument function. 

Went with unified interface for simpler DX since `worldSlug` and similar variables, as names are used everywhere, and even `props` from one component can be now plugged into `useWorldBySlug()` and similar components.


> It's not just removing reactfire, it's removing a lot of HOCs as well.

Most of which were placed as a workaround to Reactfire bugs. Not the HOCs you'd use in a regular React project though. Proper higher order components, not made as a workaround are still the way to go. Keeping these just because they are HOC... well, `withHook` makes them obsolete.

I think a more pedestrian approach of a single HOC with the logic more or less cut and paste from these components will be simpler to manage for everyone.

Reply:
Cool. Good reasoning. We should only need a few HOCs as most people want the exact same things every time: the user, the space, the world.